man/systemd-stub.xml

```diff
@@ -77,7 +77,11 @@
       <listitem><para>A <literal>.splash</literal> section with an image (in the Windows
       <filename>.BMP</filename> format) to show on screen before invoking the kernel.</para></listitem>
 
-      <listitem><para>A <literal>.dtb</literal> section with a compiled binary DeviceTree.</para></listitem>
+      <listitem><para>Zero or more <literal>.dtb</literal> sections. The stub will try to match the DeviceTrees 
+      embedded in the PE image with the DeviceTree provided by the device's firmware by comparing the first 
+      <literal>compatible</literal> string in each UKI DeviceTree with the first <literal>compatible</literal> 
+      string supplied by the firmware. If no match is found, <literal>.dtb</literal> section will be looked up by 
+      HWID. </para></listitem>
```

Comment:
So, you are using the `compatible` in those special DTBs to match against certain SMBIOS fields. That's a workaround which is fine for devices outside of any DT schema compliance. However, we need to avoid that this matching takes place also on other devices that are compliant, i.e. the normal use case. How is that achieved?

IOW: Non-standard DT matching must be an opt-in that one can control when generating a UKI.

Reply:
The matching process is the following:
1. Try to extract `compatible` from firmware-provided dtb and check against it
2. If firmware doesn't provide dtb, use SMBIOS fields

> IOW: Non-standard DT matching must be an opt-in that one can control when generating a UKI.

This can be implemented as well if needed. I see a few approaches (separate section with a boolean in the stub) or 2 stubs.

Reply:
Ok, DT compliant devices kind of always provide a DT via UEFI. That means your code will not go down the SMBIOS path when there is a DT with a compatible, just no match?

Reply:
The base description is conceptually the same, just with SMBIOS matching added. Should I remove them as primary author from here and update email for other commits?

Reply:
Ah, then it's fine to keep him. Update of email address should be good, though, to simplify traceability of the origin.